[BUG] Compile error when HAS_LCD_MENU is false in MarlinUI::external_encoder

### Bug Description

In ultralcd.h external_encoder() is only defined if HAS_LCD_MENU is true.
But in ultralcd.cpp the function external_encoder() is created when HAS_LCD_MENU is true or false.

With the right Configuration file, this can trigger the following compile error.
"Marlin/src/lcd/ultralcd.cpp:760:35: error: no 'void MarlinUI::external_encoder()' member function declared in class 'MarlinUI'"

### My Configurations

[Break_marlin_config.zip](https://github.com/MarlinFirmware/Marlin/files/4975833/Break_marlin_config.zip)

### Steps to Reproduce

1. Obtain latest marlin bugfix 2.0.x
2. Use provided Configuration.h 
3. Watch it fail to compile.

**Expected behavior:**

It should compile

**Actual behavior:**

It errors with 
Marlin/src/lcd/ultralcd.cpp:760:35: error: no 'void MarlinUI::external_encoder()' member function declared in class 'MarlinUI'

--- a/Marlin/src/lcd/ultralcd.cpp
+++ b/Marlin/src/lcd/ultralcd.cpp
@@ -1283,7 +1283,9 @@ void MarlinUI::update() {
           case encrot2: ENCODER_SPIN(encrot1, encrot3); break;
           case encrot3: ENCODER_SPIN(encrot2, encrot0); break;
         }
-        TERN_(AUTO_BED_LEVELING_UBL, external_encoder());
+        #if BOTH(HAS_LCD_MENU, AUTO_BED_LEVELING_UBL)
+          external_encoder();
+        #endif
         lastEncoderBits = enc;
       }
 

--- a/Marlin/src/lcd/ultralcd.cpp
+++ b/Marlin/src/lcd/ultralcd.cpp
@@ -753,19 +753,19 @@ void MarlinUI::quick_feedback(const bool clear_buttons/*=true*/) {
     //SERIAL_ECHOLNPAIR("Post Move with Axis ", int(axis), " soon.");
   }
 
-#endif // HAS_LCD_MENU
+  #if ENABLED(AUTO_BED_LEVELING_UBL)
 
-#if ENABLED(AUTO_BED_LEVELING_UBL)
-
-  void MarlinUI::external_encoder() {
-    if (external_control && encoderDiff) {
-      ubl.encoder_diff += encoderDiff;  // Encoder for UBL G29 mesh editing
-      encoderDiff = 0;                  // Hide encoder events from the screen handler
-      refresh(LCDVIEW_REDRAW_NOW);      // ...but keep the refresh.
+    void MarlinUI::external_encoder() {
+      if (external_control && encoderDiff) {
+        ubl.encoder_diff += encoderDiff;  // Encoder for UBL G29 mesh editing
+        encoderDiff = 0;                  // Hide encoder events from the screen handler
+        refresh(LCDVIEW_REDRAW_NOW);      // ...but keep the refresh.
+      }
     }
-  }
 
-#endif
+  #endif
+
+#endif // HAS_LCD_MENU
 
 /**
  * Update the LCD, read encoder buttons, etc.

